File: src/ProjectR.Scripting/content/scripts/spells/lua/KonSlash.lua
function GetTargetType()
    return Single
end

function GetName()
    return "Kon Slash"
end

function GetDescription()
    return "Strong Physical attack that bypasses some of the enemies armor."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 0
end

function GetDelay()
  return .5
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
    local damage = 3 * aAD - .8 * dDEF
    d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/KonSlash.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class KonSlash : SpellScriptBase
    {
        public override string Name { get { return "Kon Slash"; } }
        public override string Description { get { return "Strong Physical attack that bypasses some of the enemies armor."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] {  }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 0; } }
        public override double Delay { get { return .5; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = 3 * cAD - .8 * tDEF;
            DealDamage(damage);
        }
    }
}